### ISSUE TITLE
DROP TEMPORARY TABLE doesn't generate TEMPORARY keyword in MariaDB

### ISSUE THREAD

**Issue by BlazingMagpie** [jOOQ/jOOQ#11224]
dropTemporaryTable() and dropTemporaryTableIfExists() when using SqlDialect.MARIADB do not render "temporary" keyword in final SQL query. Both MySQL and MariaDB share this syntax and behavior and this feature was previously implemented for MySQL (#9019).

Example:
```java
DSL.using(SqlDialect.MARIADB).dropTemporaryTable(DSL.name("temp_table")).getSql().equals("drop temporary table `temp_table`")//expected;
DSL.using(SqlDialect.MARIADB).dropTemporaryTable(DSL.name("temp_table")).getSql().equals("drop table `temp_table`")//actual;

DSL.using(SqlDialect.MARIADB).dropTemporaryTableIfExists(DSL.name("temp_table")).getSql().equals("drop temporary table if exists `temp_table`")//expected;
DSL.using(SqlDialect.MARIADB).dropTemporaryTableIfExists(DSL.name("temp_table")).getSql().equals("drop table if exists `temp_table`")//actual;
```

**Comment by lukaseder** [comment]
Thank you very much for your report. Indeed, this is an oversight from #9019: https://github.com/jOOQ/jOOQ/issues/9019#issuecomment-533064597

Will fix right away and backport to 3.14.5

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.5 (#11225). Thanks again for your report!
